jmh-benchmarks/src/main/java/org/apache/kafka/jmh/assignor/StreamsStickyAssignorBenchmark.java

```diff
@@ -0,0 +1,180 @@
+
+        MetadataImage metadataImage = AssignorBenchmarkUtils.createMetadataImage(allTopicNames, partitionCount);
+
+        topologyDescriber = new TopologyMetadata(metadataImage, subtopologyMap);
+
+        taskAssignor = new StickyTaskAssignor();
+
+        Map<String, StreamsGroupMember> members = createMembers();
+        this.assignmentConfigs = Collections.singletonMap(
+            "num.standby.replicas",
+            Integer.toString(standbyReplicas)
+        );
+        this.groupSpec = StreamsAssignorBenchmarkUtils.createGroupSpec(members, assignmentConfigs);
+
+        if (assignmentType == AssignmentType.INCREMENTAL) {
+            simulateIncrementalRebalance();
+        }
+    }
+
+    private Map<String, StreamsGroupMember> createMembers() {
+        // In the rebalance case, we will add the last member as a trigger.
+        // This is done to keep the total members count consistent with the input.
+        int numberOfMembers = assignmentType.equals(AssignmentType.INCREMENTAL) ? memberCount - 1 : memberCount;
+
+        return StreamsAssignorBenchmarkUtils.createStreamsMembers(
+            numberOfMembers,
+            membersPerProcess
+        );
+    }
+
+    private void simulateIncrementalRebalance() {
+        GroupAssignment initialAssignment = new StickyTaskAssignor().assign(groupSpec, topologyDescriber);
+        Map<String, MemberAssignment> members = initialAssignment.members();
+
+        Map<String, AssignmentMemberSpec> updatedMemberSpec = new HashMap<>();
+
+        for (Map.Entry<String, AssignmentMemberSpec> member : groupSpec.members().entrySet()) {
+            MemberAssignment memberAssignment = members.getOrDefault(
+                member.getKey(),
+                new MemberAssignment(Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap())
+            );
+
+            updatedMemberSpec.put(member.getKey(), new AssignmentMemberSpec(
+                Optional.empty(),
+                Optional.empty(),
+                memberAssignment.activeTasks(),
+                memberAssignment.standbyTasks(),
+                memberAssignment.warmupTasks(),
+                member.getValue().processId(),
+                Map.of(),
+                Map.of(),
+                Map.of()
+            ));
+        }
+
+        updatedMemberSpec.put("newMember", new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Map.of(),
+            Map.of(),
+            Map.of(),
+            "process-newMember",
+            Map.of(),
+            Map.of(),
+            Map.of()
+        ));
+
+        groupSpec = new GroupSpecImpl(
+            updatedMemberSpec,
+            assignmentConfigs
+        );
+    }
+
+    @Benchmark
+    @Threads(1)
+    @OutputTimeUnit(TimeUnit.MILLISECONDS)
+    public void doAssignment() {
+        taskAssignor.assign(groupSpec, topologyDescriber);
```

Comment:
This should either return the calculated `GroupAssignment` object or use a [JMH Blackhole](https://javadoc.io/doc/org.openjdk.jmh/jmh-core/1.23/org/openjdk/jmh/infra/Blackhole.html) to consume it so the JIT compiler thinks it's used elsewhere and doesn't get optimized away as dead code.  Probably makes more sense to just update the return type and add `return taskAssignor.assign(xxx, yyy)`

Reply:
Good point! I get a compiler warning when just returning the value, So I added a blackhole.